### PR TITLE
Add types for `cursor()` and `controller()` in mathquill.d.ts

### PR DESCRIPTION
**Pull request by nishasy**
## Summary:

Currently, the type for `cursor()` and `controller()` on `Mathfield` are `any`. We want them to be properly typed.

In this PR:
- Add types for `controller()` and `cursor()` in build/mathquill.d.ts.
- Remove these in src/mathquill.d.ts since they're not being used here, and they'd mess up types in other places.

Issue: https://khanacademy.atlassian.net/browse/LC-1659

## Test plan:

- Copy all the new contents of [build/mathquill.d.ts](https://github.com/Khan/mathquill/blob/main/build/mathquill.d.ts)
- Paste them in khan/perseus/node_modules/mathquill/build/mathquill.d.ts (local file)
- Run `yarn typecheck` in the Perseus repo
- Confirm there are no type errors

**Testing note:**
This requires changing line 24 in [khan/perseus/packages/math-input/src/components/input/__tests__/test-math-wrapper.ts](https://github.com/Khan/perseus/blob/e853bbbff4f9ee0886801ad48e9af5d7e1e3d226/packages/math-input/src/components/input/__tests__/test-math-wrapper.ts#L24) from
```
return selection.ends[-1][-1] === 0 && selection.ends[1][1] === 0;
```
to 
```
return (
    selection.getEnd([-1][-1]) === 0 &&
    selection.getEnd([1][1]) === 0
);
```

### CODE REVIEW

File: build/mathquill.d.ts [hunk 1]
```diff
@@ -22,6 +22,47 @@ declare namespace MathQuill {
...
+  
+      backspace(): Controller;
+      seek(targetElm, clientX, _clientY): Controller;
+    }
+  
+    interface ControllerRoot {
+      controller: Controller;
+      cursor?: Cursor;
+    }
+
```
Comment (nishasy) [inline build/mathquill.d.ts:65]: **question**: I put these inside the `v3` namespace. Should they be outside, maybe around line 9 instead?

File: build/mathquill.d.ts [hunk 2]
```diff
@@ -22,6 +22,47 @@ declare namespace MathQuill {
...
+      id: number;
+      parent: NodeBase;
+    }
+  
+    interface MQSelection {
+      id: number;
+      getEnd(dir: number): number;
+    }
+  
+    interface NodeBase extends MQNode {
```
Comment (nishasy) [inline build/mathquill.d.ts:35]: `extends MQNode` does not feel right to me here, but it's the only thing that worked with the rest of the types in Perseus.

**Review by nedredmond — CHANGES_REQUESTED** [review]
THanks Nisha! After merging my last PR, these would be overwritten on the next build, so we should explore some more options.

I'm okay with the cheap fix, which is adding these types to the namespace either in Build or in Perseus. However, I would like to try to pair on getting real exportable types.

Here's my text from the DM:

> Hey, whenever you're back we should think a bit more about the controller and cursor types, since the build step will now overwrite that file (or will when my PR to avoid manual fixing of build files is done).
I think it's valuable to be able to have one source of truth for code and types, so if you agree, I'd like to find a way to make our custom types work with that.
We have some choices:
- Add a dist-only declarations file that adds the types to the build.
  - This ranks 2nd for me. Makes things more complex.
- Inject our types via MakeFile.
  - Ranks last. Fragile, weird.
- Try again to incorporate it into the src declarations file.
  - I think this is the best choice, if we can manage it. I see two paths here:
    - This weekend I saw that we can declare class to create a class type. Maybe we can re-implement the whole thing after all?
    - I think the problem we were seeing last week was the partial type overriding the class. There must be a simple way to differentiate the results of cursor() from _controller.cursor(), and controller from _controller. Maybe if we gave them different names? CursorType, ICursor-- something like that.

**Comment by nishasy** [comment]
Abandoning this because this approach would make it difficult to keep the build/ and src/ mathquill.d.ts files in sync.

**Pull request closed without merging**